### ISSUE TITLE
topic_model.visualize_barchart() custom_labels overrides original labels

### ISSUE THREAD

**Issue by fatihbozdag** [MaartenGr/BERTopic#1168]
Greetings all, I have a problem with visualizing topics.

```
topics, probs = topic_model.fit_transform(docs)

topic_labels = topic_model.generate_topic_labels(nr_words=5,
                                                 topic_prefix=False,
                                                 word_length=10,
                                                 separator=", ")
topic_model.set_topic_labels(topic_labels)

topic_model.visualize_barchart(top_n_topics = 10,**custom_labels=True**)
```

with custom_labels = True, chart becomes unreadable. 
![Screenshot 2023-04-06 at 23 14 19](https://user-images.githubusercontent.com/59308858/230485324-dd975642-dc62-4527-9687-d46a029a3395.png)

What is the problem?

**Comment by MaartenGr** [comment]
That is actually to be expected. The title of a graph is of fixed size (font-wise) and as such when you create very long custom labels they will become a bit too much for the graph. I am not too sure whether it would be possible to automatically resize the font based on the length of the titles in Plotly. It might be possible outside of Plotly by using a monospaced font, calculating its size, comparing it with the width of a graph, and then changing the font size but I am afraid that introduces a lot of unnecessary complexity.

**Comment by fatihbozdag** [comment]
> That is actually to be expected. The title of a graph is of fixed size (font-wise) and as such when you create very long custom labels they will become a bit too much for the graph. I am not too sure whether it would be possible to automatically resize the font based on the length of the titles in Plotly. It might be possible outside of Plotly by using a monospaced font, calculating its size, comparing it with the width of a graph, and then changing the font size but I am afraid that introduces a lot of unnecessary complexity. 

Well actually it would be better if we have the option to replace new labels with the old ones. For now I’ll try to find an alternative way. 

Meanwhile I have another question, not relevant to the current. Is there a way to integrate entropy/distribution data into the topic frequency? Particularly working with a series of documents from different classes (in my case they are not actually classes but a kind of doc_id representing data source), it would be great to know how a certain topic is dispersed over documents. For instance, I have five different data sources which I merged into one on Pandas and each document gets a doc_id. Not all sources have equal number of documents. I guess it is possible that one topic might be the most frequent one if it is repeated over in one data source. In this case, regardless of others, BERTopic would assume it is the prominent one. I did indeed wrote a simple snippet to calculate entropy over ‘probs’ and merged with frequency data. And the question is, when leveraging can we also provide entropy/distribution calculation as parameter to reveal topics? Not just the most frequent one but also both the most frequent one with the highest entropy. Indeed, I can simply drop the documents where the most frequent topic is centered but then it feels like invalidation of initial calculations. 

For instance,

Sources A (100 docs) B (200docs), C (75 docs), D (75 docs), E (50 docs). Therefore the most frequent Topic 0 in merged dataframe as it is the most common in source A and B but not others. Topic 5 is the sixth most frequent one yet has higher entropy score than Topic 0. So how do you think I should handle the case? 

P.S if required I can start a new topic. Also I’ll add my entropy code as soon as possible.

**Comment by fatihbozdag** [comment]
Very well then, So here is my way to calculate entropy for those you might need it.

```
import numpy as np
doc_topic_matrix = np.array(probs)
normalized_doc_topic_matrix = doc_topic_matrix / doc_topic_matrix.sum(axis=1, keepdims=True)
# Calculate topic entropy
topic_entropy = (-normalized_doc_topic_matrix * np.log2(normalized_doc_topic_matrix)).sum(axis=0)

# Create a DataFrame with topic entropy values
entropy_df = pd.DataFrame({'Topic': range(len(topic_entropy)), 'Entropy': topic_entropy})

# Sort the DataFrame by entropy in descending order
sorted_entropy_df = entropy_df.sort_values('Entropy', ascending=False)

sorted_entropy_df
```

```
# Merge Topic Freqs with Entropy DataFrame Display the sorted.
topic_freq = topic_model.get_topic_freq()
topic_freq = topic_freq[topic_freq['Topic'] != -1]

topic_freq['Entropy'] = sorted_entropy_df['Entropy']
topic_freq
```

**Comment by MaartenGr** [comment]
Thanks for sharing the code. This is also relevant for users that split up their texts to prevent token limits and/or have documents that each contain a wide distribution of topics.
